app/src/main/java/kth/jjve/xfran/SaveResultsActivity.java

```diff
@@ -40,53 +55,103 @@ protected void onCreate(Bundle savedInstanceState) {
+        Log.i(LOG_TAG, "workout read: " + workout);
 
         // Build a string with exercises
-        StringBuilder exercises = new StringBuilder();
-        ArrayList<String> exercisesArray = mWorkout.getExercises();
-        for (String s : exercisesArray){
-            exercises.append(s).append("\n");
-        }
+        String exercises = WorkoutUtils.buildExerciseString(workout);
 
         // Fill the UI with workout info
-        mName.setText(mWorkout.getTitle());
-        mDescription.setText(mWorkout.getType());
-        mExercises.setText(exercises.toString());
-
+        mName.setText(workout.getTitle());
+        mDescription.setText(workout.getDescription());
+        mExercises.setText(exercises);
         workoutItemSaveButton.setVisibility(GONE);
         workoutItemPlanButton.setVisibility(GONE);
 
-        //TODO finish this activity --> work in progress
+        //change score_type text according to the current workout
+        String scoreType = ResultUtils.setScoreType(workout);
+        mScoreType.setText(scoreType);
+
+        //change edit_score hint according to the current workout
+        String scoreTypeHint = ResultUtils.setScoreTypeHint(workout);
+        mScore.setHint(scoreTypeHint);
+
+        //change edit_score digits according to the current workout
+        String scoreDigits = ResultUtils.setScoreDigits(workout);
+        mScore.setKeyListener(DigitsKeyListener.getInstance(scoreDigits));
+
+        // Default workout date set to current date
+        todayLocalDate = LocalDate.now();
+        mDate.setText(todayLocalDate.toString());
 
     }
 
     @Override
-    protected void onResume(){
+    protected void onResume() {
         super.onResume();
         navigationView.setCheckedItem(R.id.nav_workouts);
     }
 
-    public void saveResult(){
-        // TODO save result object --> work in progress
-        Toast.makeText(this, "Save workout in development", Toast.LENGTH_SHORT).show();
+    public void saveResult() {
+        //get all filled fields
+        try {
+            LocalDate date = LocalDate.parse(mDate.getText());
+            boolean scaled = mScaledSwitch.isChecked();
+            String score = String.valueOf(mScore.getText()); // Todo: deal better with different types of scores
+            Integer rating = Math.round(mScoreSeekBar.getProgress());
+            String comments = mComments.getText().toString();
+
+            //error message if score is not filled
+            if (TextUtils.isEmpty(score)) {
+                mScore.setError("Score is required");
+                return;
+            }
+            //error message if score has wrong format
+            if (ResultUtils.isWrongScore(workout, score)) {
+                mScore.setError("Score is required");
+                return;
+            }
+            //error message user enters date in the future
+            if (date.isAfter(LocalDate.now())) {
+                mDate.setError("You are not a clairvoyant");
```

Comment:
:D :D :D love it

Reply:
originally was a cheater but thought it was too aggressive hahaha